ee/clickhouse/client.py

```diff
@@ -74,6 +85,18 @@ def async_execute(query, args=None):
         connections_max=100,
     )
 
+    def cache_sync_execute(query, args=None, redis_client=None, ttl=CACHE_TTL):
```

Comment:
When will we use this vs sync_execute?

Reply:
Whenever you want to cache the response. There will definitely be cases where you don't want to cache the response and want the most up to date result. For instance when I grab the most recent event timestamp for monitoring.

Reply:
Rather than having a separate function it might be nicer to just allow you to pass through `cached=True` to sync_execute?

Reply:
I don't have a strong opinion one way or another for that, do you? We'd want two kwargs: `cached=True` and `ttl=<some_time>` which you'd want to pass together which feels more brittle to me

Reply:
I kind of like having it be very clear that this is cached so there maybe unexpected results. As well as a separate set of params required that are unique to the caching. Generally I like functions with a smaller parameter footprint too. So I guess I do kind of have an opinion on this :P